generator.py

```diff
@@ -92,30 +94,32 @@ def phi(s: [str], n: int, v: [str], f: [str], p: [str], g: str):
     # 1.state='NY'
     for i in f:
         aggregate_function, gv_num, aggregate_attribute = i.split("_")
-        predicate = p[int(gv_num)]
+        predicate = p[int(gv_num) -1 ]
```

Comment:
We are doing a default grouping variable for group by attributes so you cannot do -1 here it might lead to a bug.